HerPortal.BusinessLogic/Services/CsvFileService/CsvFileService.cs

```diff
@@ -58,7 +58,31 @@ public async Task<IEnumerable<CsvFileData>> GetFileDataForUserAsync(string userE
             .OrderByDescending(f => new DateOnly(f.Year, f.Month, 1))
             .ThenBy(f => LocalAuthorityData.LocalAuthorityNamesByCustodianCode[f.CustodianCode]);
     }
-    
+
+    // Page number starts at 1
+    public async Task<PaginatedFileData> GetPaginatedFileDataForUserAsync(
+        string userEmailAddress,
+        List<string> custodianCodes,
+        int pageNumber,
+        int pageSize)
+    {
+        var allFileData = (await GetFileDataForUserAsync(userEmailAddress)).ToList();
+        var filteredFileData = allFileData
+            .Where(cfd => custodianCodes.Count == 0 || custodianCodes.Contains(cfd.CustodianCode))
+            .ToList();
+
+        var maxPage = filteredFileData.Count / pageSize + (filteredFileData.Count % pageSize == 0 ? 0 : 1);
```

Comment:
You could simplify slightly to `(filteredFileData.Count - 1) / pageSize + 1` (though arguably your logic is easier to follow)

Reply:
I knew there had to a simpler way, but forgot to come back to this :)